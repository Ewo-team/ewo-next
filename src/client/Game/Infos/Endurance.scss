.Infos__Endurance {
    grid-area: endu;
}
