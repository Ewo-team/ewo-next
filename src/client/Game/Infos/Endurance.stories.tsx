// import { action } from '@storybook/addon-actions';
import { number, withKnobs } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import { EnduranceComponent } from '.';

const stories = storiesOf('Infos/Endurance', module);

stories.addDecorator(withKnobs);

stories.add('basic Endurance', () => <EnduranceComponent
  character={{
    minutes: number('Minute', 0, { range: true, min: 0, max: 59, step: 1 }),
  }}
/>);

stories.add('Endurance is full', () => <EnduranceComponent character={{}} />);
