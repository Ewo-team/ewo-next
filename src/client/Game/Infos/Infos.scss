.Infos {
    grid-area: Infos;
    display: grid;
    grid-template: "position screenname" 1fr
                    ". endu" 1fr /
                    1fr 1fr;
}
