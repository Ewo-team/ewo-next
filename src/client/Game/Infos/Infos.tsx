/**
 * @module Client.Game.Infos
 * Information module
 */

import * as React from 'react';
import { connect } from 'react-redux';
import { Endurance, Position, ScreenName } from '.';

export const InfosComponent = () => {
  return (
    <div className="Infos Game__Container Header">
      <Position />
      <ScreenName />
      <Endurance />
    </div>
  );
};

export const Infos = connect()(InfosComponent);
