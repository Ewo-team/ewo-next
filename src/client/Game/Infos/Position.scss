.Infos__Position {
    grid-area: position;
}