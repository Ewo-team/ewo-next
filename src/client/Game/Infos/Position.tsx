/**
 * @module Client.Game.Infos
 * Position module
 */

import { IStateFrontend } from '@client/reducers';
import { getSelectedCharacter } from '@client/selector';
import { CharacterFrontend, Plan } from '@models';
import * as React from 'react';
import { connect } from 'react-redux';
import { Plans } from '@engine/resources';

export interface PositionProps {
  character: CharacterFrontend;
}

export const PositionComponent = (props: PositionProps) => {

  if (props.character === undefined) {
    return null;
  }


  if (props.character.coord) {
    const { plan, x, y } = props.character.coord;
    const planObj: Plan = Plans[plan];
    return <div className="Infos__Position">Sur <a href="gps">{planObj.name}</a> en X = {x} <b>| </b>Y = {y}</div>;
  }
  return <div className="Infos__Position">Nullepart</div>;
};

const mapStateToProps = (state: IStateFrontend) => ({
  character: getSelectedCharacter(state),
});

export const Position = connect(
  mapStateToProps,
)(PositionComponent);
