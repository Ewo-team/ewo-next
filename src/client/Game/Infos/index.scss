@import './Endurance.scss';
@import './Position.scss';
@import './ScreenName.scss';

.Infos {
    grid-area: Infos;
    display: grid;
    grid-template: "position screenname" 1fr
                    ". endu" 1fr /
                    1fr 1fr;
}
