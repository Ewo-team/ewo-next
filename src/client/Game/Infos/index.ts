export * from './Endurance';
export * from './Infos';
export * from './Position';
export * from './ScreenName';
