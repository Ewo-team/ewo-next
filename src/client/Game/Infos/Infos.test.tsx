import React from 'react';
import renderer from 'react-test-renderer';
//import configureStore from 'redux-mock-store';
import { EnduranceComponent, ScreenNameComponent } from '.';
import {
  characterFrontendDeadMat2,
  characterFrontendDefaultMat1,
  characterFrontendFullMat3,
} from '../../../../test/mock/character';
import { CharacterFrontend } from '../../../engine/models';
import { Plans } from '../../../engine/resources';
import { refreshCharacters, setSelectedCharacter } from '../../actions';
import { Provider } from '../../provider';
import { createStore } from '../../store';
import { Infos } from './Infos';
import { PositionComponent } from './Position';
import { frontendInitialState } from '../../../../test/mock/frontendStore';

// tslint:disable
const mockedDate = new Date(2019, 0, 1, 10, 0, 0, 0);
const _Date = Date;
(global as any).Date = jest.fn(() => mockedDate); // minute at 0
global.Date.UTC = _Date.UTC;
global.Date.parse = _Date.parse;
global.Date.now = _Date.now;

//const mockStore = configureStore();

describe('UI Infos', () => {


  describe('Info connected component', () => {

    const store = createStore(undefined, frontendInitialState);

    const def: CharacterFrontend = characterFrontendDefaultMat1
    const dead: CharacterFrontend = characterFrontendDeadMat2;
    const full: CharacterFrontend = characterFrontendFullMat3;

    it('Default character', () => {

      renderer.act(() => {
        store.dispatch(setSelectedCharacter(def.mat));
      });

      const component = renderer.create(
        <Provider store={store}>
          <Infos />
        </Provider>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('default character');
    });

    it('Dead character', () => {

      renderer.act(() => {
        store.dispatch(setSelectedCharacter(dead.mat));
      });

      const component = renderer.create(
        <Provider store={store}>
          <Infos />
        </Provider>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('dead character');
    });

    it('Full endurance character', () => {

      renderer.act(() => {
        store.dispatch(setSelectedCharacter(full.mat));
      });

      const component = renderer.create(
        <Provider store={store}>
          <Infos />
        </Provider>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('full character');
    });

    it('no character', () => {

      renderer.act(() => {
        store.dispatch(setSelectedCharacter(undefined));
      });

      const component = renderer.create(
        <Provider store={store}>
          <Infos />
        </Provider>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('no character');
    });
  })

  describe('Endurance', () => {

    it('should render with minute at 10', () => {

      const character = {
        minutes: 10,
      } as CharacterFrontend;

      const component = renderer.create(
        <EnduranceComponent
          character={character}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('minutes 10');
    });

    it('should render with minute at 45', () => {

      const character = {
        minutes: 45,
      } as CharacterFrontend;

      const component = renderer.create(
        <EnduranceComponent
          character={character}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('minutes 45');
    });

    it('should render with no minute', () => {
      const character = {} as CharacterFrontend;

      const component = renderer.create(

        <EnduranceComponent
          character={character}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('no minutes');
    });

    it('should render with no character', () => {

      const component = renderer.create(

        <EnduranceComponent
          character={undefined}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('no character');
    });

  });

  describe('Position', () => {
    it('should render with position in 0 / 0 on Althian', () => {

      const althian = Plans.althian.id;

      const character = {
        coord: {
          plan: althian,
          x: 0,
          y: 0,
        },
      } as CharacterFrontend;

      const component = renderer.create(
        <PositionComponent
          character={character}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('position 0 / 0 on althian');
    });

    it('should render with no position', () => {
      const character = {} as CharacterFrontend;

      const component = renderer.create(

        <PositionComponent
          character={character}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('no position');
    });

    it('should render with no character', () => {

      const component = renderer.create(

        <PositionComponent
          character={undefined}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('no character');
    });
  });

  describe('ScreenName', () => {
    it('should render with name Jest, mat 1, grade 2g3', () => {


      const character = {
        name: 'Jest',
        mat: 1,
        grade: {
          major: 2,
          minor: 3,
        },
      } as CharacterFrontend;

      const component = renderer.create(
        <ScreenNameComponent
          character={character}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('jest mat 1 G2g3');
    });

    it('should render with no character', () => {

      const component = renderer.create(
        <ScreenNameComponent
          character={undefined}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot('no character');
    });
  });
});
