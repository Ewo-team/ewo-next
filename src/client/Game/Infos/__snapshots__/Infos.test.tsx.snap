// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Infos Endurance should render with minute at 10: minutes 10 1`] = `
<div
  className="Infos__Endurance progress"
>
  <div
    aria-valuemax={60}
    aria-valuemin={1}
    aria-valuenow={50}
    className="progress-bar progress-bar-striped"
    role="progressbar"
    style={
      Object {
        "width": "83%",
      }
    }
  >
    10
     minutes
  </div>
</div>
`;

exports[`UI Infos Endurance should render with minute at 45: minutes 45 1`] = `
<div
  className="Infos__Endurance progress"
>
  <div
    aria-valuemax={60}
    aria-valuemin={1}
    aria-valuenow={15}
    className="progress-bar progress-bar-striped"
    role="progressbar"
    style={
      Object {
        "width": "25%",
      }
    }
  >
    45
     minutes
  </div>
</div>
`;

exports[`UI Infos Endurance should render with no character: no character 1`] = `null`;

exports[`UI Infos Endurance should render with no minute: no minutes 1`] = `
<div
  className="Infos__Endurance"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="progress-bar bg-success"
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`UI Infos Info connected component Dead character: dead character 1`] = `
<div
  className="Infos Game__Container Header"
>
  <div
    className="Infos__Position"
  >
    Nullepart
  </div>
  <div
    className="Infos__ScreenName"
  >
    <strong>
      Dead Character
    </strong>
     (
    2
    ) | Grade 
    2
     galon 
    1
  </div>
  <div
    className="Infos__Endurance"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={100}
      className="progress-bar bg-success"
      role="progressbar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`UI Infos Info connected component Default character: default character 1`] = `
<div
  className="Infos Game__Container Header"
>
  <div
    className="Infos__Position"
  >
    Sur 
    <a
      href="gps"
    >
      Althian
    </a>
     en X = 
    15
     
    <b>
      | 
    </b>
    Y = 
    -5
  </div>
  <div
    className="Infos__ScreenName"
  >
    <strong>
      Character in action
    </strong>
     (
    1
    ) | Grade 
    2
     galon 
    1
  </div>
  <div
    className="Infos__Endurance progress"
  >
    <div
      aria-valuemax={60}
      aria-valuemin={1}
      aria-valuenow={43}
      className="progress-bar progress-bar-striped"
      role="progressbar"
      style={
        Object {
          "width": "72%",
        }
      }
    >
      17
       minutes
    </div>
  </div>
</div>
`;

exports[`UI Infos Info connected component Full endurance character: full character 1`] = `
<div
  className="Infos Game__Container Header"
>
  <div
    className="Infos__Position"
  >
    Sur 
    <a
      href="gps"
    >
      Althian
    </a>
     en X = 
    1
     
    <b>
      | 
    </b>
    Y = 
    2
  </div>
  <div
    className="Infos__ScreenName"
  >
    <strong>
      Full Endurance Character
    </strong>
     (
    3
    ) | Grade 
    2
     galon 
    1
  </div>
  <div
    className="Infos__Endurance"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={100}
      className="progress-bar bg-success"
      role="progressbar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`UI Infos Info connected component no character: no character 1`] = `
<div
  className="Infos Game__Container Header"
/>
`;

exports[`UI Infos Position should render with no character: no character 1`] = `null`;

exports[`UI Infos Position should render with no position: no position 1`] = `
<div
  className="Infos__Position"
>
  Nullepart
</div>
`;

exports[`UI Infos Position should render with position in 0 / 0 on Althian: position 0 / 0 on althian 1`] = `
<div
  className="Infos__Position"
>
  Sur 
  <a
    href="gps"
  >
    Althian
  </a>
   en X = 
  0
   
  <b>
    | 
  </b>
  Y = 
  0
</div>
`;

exports[`UI Infos ScreenName should render with name Jest, mat 1, grade 2g3: jest mat 1 G2g3 1`] = `
<div
  className="Infos__ScreenName"
>
  <strong>
    Jest
  </strong>
   (
  1
  ) | Grade 
  2
   galon 
  3
</div>
`;

exports[`UI Infos ScreenName should render with no character: no character 1`] = `null`;
